plugins/ssl/letsencrypt_weekly

```diff
@@ -49,13 +107,42 @@ elif [ "$1" = "config" ] ; then
 	echo "graph_vlabel Number of certificates"
 	echo "graph_category security"
 	echo "letsencrypt_weekly.label Letsencrypt certificates last week"
-	print_warning "letsencrypt_weekly"
-	print_critical "letsencrypt_weekly"
+	for domain in $(get_registered_domains); do
+		key=${domain//[-.]/_}
+		echo "${key}_weekly.label $domain"
+		print_warning "${key}_weekly"
+		print_critical "${key}_weekly"
+		echo "${key}_renewal_weekly.label $domain renewals"
+	done
 elif [ "$1" = "" ] ; then
-	if existing_certs=$(find /etc/letsencrypt/csr/ -mtime -7 -type f 2>/dev/null); then
-		value=$(echo "$existing_certs" | wc -l)
+	if existing_renewal_requests=$(get_files_and_domains); then
+		value=$(echo "$existing_renewal_requests" | grep -v '^$' -c)
 	else
 		value="U"
 	fi
 	echo "letsencrypt_weekly.value $value"
+	values=""
+	for domain in $(get_registered_domains); do
+		key=${domain//[-.]/_}
+		if [ "$values" != "" ] ; then
+			values="$values\n${key}_weekly.value 0\n${key}_renewal_weekly.value 0"
+		else
+			values="${key}_weekly.value 0\n${key}_renewal_weekly.value 0"
+		fi
+	done
+	while read -r file_domain; do
+		file=${file_domain% *}
+		domain=${file_domain#* }
+		registered_domain_key=$(echo "$domain" | sed 's/.*\.\([a-z0-9-]\+\.[a-z]\+\)/\1/;s/[-.]/_/g')
+		previous_certs=$(find "/etc/letsencrypt/archive/$domain" -name 'cert*.pem' -not -cnewer "$file" | wc -l)	
+		if [ "$previous_certs" -gt 0 ] ; then
+			value_key="${registered_domain_key}_renewal_weekly.value "
+		else
+			value_key="${registered_domain_key}_weekly.value "
+		fi
+		old_value=$(echo -e $"$values" | grep "$value_key" | sed 's/.* //g')
+		value=$((old_value + 1))
+		values=${values//$value_key$old_value/$value_key$value}
+	done < <(get_files_and_domains)
```

Comment:
Is the first `<` relevant?
(I do not use bash too often, thus I may be wrong)

Reply:
yes this is necessary, (in my words maybe not the best explanation):
* the first `<` prepares the while loops stdin to be provided by the given file(descriptor)
* the second `<(...)` is process substitution which prepares a filedescriptor that can only be read by your process and is fed by the command given (`echo <(echo "ok")` displays `/dev/fd/63`, `cat <(echo "ok")` gives you `ok`)

This allows the main process to hold and change the variables while the following would not allow changing variables in the while loop (because the while loop would run in a subprocess):
```
get_files_and_domains | while read -r line; do
    # cannot change parent variables here (running in subprocess)
    echo "$line"
done

while read -r line; do
    # can change variables here (running in main process)
    echo "$line"
done < <(get_files_and_domains)
```